troubleshooting.md

```diff
@@ -49,13 +49,18 @@ The standard library headers are often found relative to the compiler.
 Try to configure your project with an absolute path to the compiler
 (`/usr/bin/gcc`, not `gcc`).
 The logs (with `--log=verbose`) can tell you the `cc1` command, including which
-paths were searched for the standard library. You can compare this to the output
+paths searched for the standard library. You can compare this to the output
 of `clang -### <args>`.
 
 If you're using an unusual compiler (e.g. a cross-compiler for a different
 platform, or MinGW on Windows) you may want to pass `--query-driver=/path/to/mygcc`
 or `--query-driver=/path/to/mygcc,/path/to/myg++` when using C++ to allow clangd
-to extract the include paths from it directly.
+to extract the include paths from it directly. 
+
+It is preferable to use `--query-driver` over [configuration files](config#files)
```

Comment:
Suggested revision to the wording:

"It is recommended to use `--query-driver` over specifying system include paths manually using [configuration files](config#compileflags) as the latter can be tricky to get right (e.g. the order of include paths can be important). Generally the value of `--query-driver` should match the command emitted by the build system in `compile_commands.json`, e.g. `/usr/bin/c++`".

(Note, I removed "for most" after the mention of `/usr/bin/c++`, because that has not been my experience. The path that ends up in `compile_commands.json` tends to vary widely, from `/path/to/clang++` through `/path/to/g++` to `/path/to/c++`, and the path part can also be all sorts of things different from `/usr/bin`.)

Reply:
> as the latter can be tricky to get right

"as getting the latter right can be tricky" seems more correct

> (e.g. the order of include paths can be important)

i dont see a reason to use modality, maybe _is_ is better than _can be_
"(the order of include paths is important, for instance)" seems clearer to me

> e.g. `/usr/bin/c++`

", such as /usr/bin/c++." might be better than relying on the _example given_ convention for foreigners reading?